gcc/rust/rust-session-manager.cc

```diff
@@ -1113,3 +1186,32 @@ TargetOptions::enable_implicit_feature_reqs (std::string feature)
  *  - code generation
  *  - link */
 } // namespace Rust
+
+#if CHECKING_P
+namespace selftest {
+void
+rust_crate_name_validation_test (void)
```

Comment:
You need to call this function from the `rust-lang.cc` file. It contains the hook and entrypoint of the unit tests for our frontend. Simply add a call to `rust_crate_name_validation_test ()` in the `run_rust_tests()` function

Reply:
Function call added.